CodeEdit/AppDelegate.swift

```diff
@@ -224,7 +245,38 @@ final class AppDelegate: NSObject, NSApplicationDelegate, ObservableObject {
 
     @objc
     func documentController(_ docController: NSDocumentController, didCloseAll: Bool, contextInfo: Any) {
-        NSApplication.shared.reply(toApplicationShouldTerminate: didCloseAll)
+        if didCloseAll {
+            terminateTasks()
+            terminateLanguageServers()
+        }
+    }
+
+    /// Terminates running language servers. Used during app termination to ensure resources are freed.
+    private func terminateLanguageServers() {
+        Task {
+            await withTaskGroup(of: Void.self) { group in
+                await lspService.languageClients.forEach { (key, value) in
+                    group.addTask {
+                        do {
+                            try await value.shutdown()
+                        } catch {
+                            self.logger.error("Shutting down \(key.languageId.rawValue): Error \(error)")
+                        }
+                    }
+                }
+            }
+            await MainActor.run {
+                NSApplication.shared.reply(toApplicationShouldTerminate: true)
+            }
+        }
+    }
+
+    /// Terminates all running tasks. Used during app termination to ensure resources are freed.
+    private func terminateTasks() {
+        let documents = CodeEditDocumentController.shared.documents.compactMap({ $0 as? WorkspaceDocument })
+        documents.forEach { workspace in
+            workspace.taskManager?.terminateActiveTask()
```

Comment:
All tasks should be terminated, not just the active one. `stopAllTasks()` currently sends a `SIGINT`, but we should consider sending a `SIGTERM`.
```suggestion
            workspace.taskManager?.stopAllTasks()
```